### DISCUSSION TITLE
Control system on Array variables

### DISCUSSION THREAD

**Discussion by pengfejeac** [idaholab/moose#25726] [Q&A General]
Hi there! 

Im trying to apply control system to a Neumann boundary and I tried applied RealFunctionControl + RealControlParameterReporter with a standard variable and it works well. But when I apply the same logic to an array variable, it failed. Im aware that there are differences between scalar variablr and array variable even if there is only one component in array variable, but I cant find a better set of functions to use. Can you guys help me out?

Thanks!

**Comment by lindsayad** [comment]
Hi @pengfejeac the difference is that `value` is declared as controllable in `NeumannBC` but not in `ArrayNeumannBC`. Would you feel comfortable modifying the code to make that controllable?

**Reply by lindsayad** [reply]
Oh sorry, it will not be that simple. I didn't realize at first that `value` is a `RealEigenVector` parameter so objects like `RealFunctionControl` won't work with it. @zachmprince @YaqiWang has anybody done anything with controlling `RealEigenVector` parameters to your knowledge?

**Reply by pengfejeac** [reply]
Hi @lindsayad, thanks for your response! I did some small experiment about this lately. If the RealEigenVector contains only a few values, then I can modified the source file of boundary condition and add a few controllable Real variable to construct the RealEigenVector. But this method will not very ideal if the RealEigenVector is relatively large. So it's not quite helpful.
